sway/input/keyboard.c

```diff
@@ -449,6 +450,20 @@ static void handle_keyboard_modifiers(struct wl_listener *listener,
 	determine_bar_visibility(modifiers);
 }
 
+void add_kbd_to_view(struct sway_container *con, void *data) {
+	if (con->node.type != N_CONTAINER) {
+		return;
+	}
+
+	struct layout_per_kbd *lpk = malloc(sizeof(struct layout_per_kbd));
```

Comment:
Why?

Reply:
It's just too easy to forget to initialize one field

Reply:
It protects against a situation where we add a property to the struct later, forget to initialise it to null and therefore have invalid pointers running around. Using `calloc` for structs is a good habit to get into, and one that we're applying across the whole project.

`malloc` is fine for strings though.

Reply:
> It's just too easy to forget to initialize one field

Okay, you're maintainers, so I'll replace this one *(I'm gonna address comments a bit later, when I'll be back home)*.

However I want to note that I strictly against C-style initialization of structs to "all zeroes". All it does is hides bugs — exactly when someone forgets to initialize a field. You need to understand that "all zeroes" is not an initialized struct, because all it does is sets it to an arbitrary value. It will only be initilized, when you assign the fields explicitly — with designated initializers or otherwise.

Reply:
Disagree. Zero is a special value for pointers as well as for almost all values.

Reply:
> Disagree. Zero is a special value for pointers as well as for almost all values.

Not for file descriptors, and not necessarily for enums as well as not necessarily for number types.